deeplearning4j-core/src/main/java/org/deeplearning4j/datasets/savers/DataSetSaver.java

```diff
@@ -0,0 +1,111 @@
+package org.deeplearning4j.datasets.savers;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FilenameUtils;
+import org.nd4j.linalg.dataset.api.DataSet;
+import org.nd4j.linalg.dataset.api.iterator.DataSetIterator;
+
+import java.io.File;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * This class can be helpful when users try to save datasets in parallel manner
+ *
+ * the default number of threads is cores * 2
+ * (but you can set it manually using setNumThreads())
+ *
+ * Created by kepricon on 17. 4. 19.
+ */
+@Slf4j
+public class DataSetSaver {
+    private static int NUM_CORES = Runtime.getRuntime().availableProcessors();
+    private static int numThreads = NUM_CORES * 2;
+    private static int QUEUE_CAPACITY = 50;
+    private static ThreadPoolExecutor taskExecutor = null;
+
+    public static void setNumThreads(int numThreads) {
+        DataSetSaver.numThreads = numThreads;
+    }
+
+    public static void setQueueCapacity(int qCapacity) {
+        DataSetSaver.QUEUE_CAPACITY = qCapacity;
+    }
+
+    private static void initExecutors() {
+        taskExecutor = new ThreadPoolExecutor(numThreads, numThreads,
```

Comment:
In this case it might be easier to use something like Executors.newFixedPoolExecutor() with custom Thread factory. Like this: https://github.com/deeplearning4j/deeplearning4j/blob/4bafd6c7b108d1c1d03d97bf6b3ce4f51c4da304/deeplearning4j-core/src/main/java/org/deeplearning4j/datasets/datavec/RecordReaderDataSetIterator.java#L208-L208

This way allows to avoid shutdown, and guarantees threads to be daemonic, so not blocking app exit.

P.s. I'm NOT requesting change here. It's jfyi.